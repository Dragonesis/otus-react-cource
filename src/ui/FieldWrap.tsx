import React, { FC, PropsWithChildren } from 'react'

import styled from '@emotion/styled'

export interface FieldWrapProps extends PropsWithChildren<unknown> {
  isOffset?: boolean
}

export const FieldWrap: FC<FieldWrapProps> = ({ isOffset, children }) => {
  return (
    <Core isOffset={isOffset}>
      {children}
    </Core>
  )
}


export interface CoreProps {
  isOffset?: boolean
}
export const Core = styled.label<CoreProps>`
  display: block;
  width: 100%;
  margin-bottom: ${({ isOffset }) => isOffset && '16px'};
`
