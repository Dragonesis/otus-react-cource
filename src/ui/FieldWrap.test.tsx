import React from 'react'
import { render } from '@testing-library/react'

import { FieldWrap } from './FieldWrap'

describe('### FieldWrap', () => {
  test('FieldWrap renders correctly', () => {
    const { asFragment } = render(<FieldWrap />)
    expect(asFragment()).toMatchSnapshot()
  })
})
