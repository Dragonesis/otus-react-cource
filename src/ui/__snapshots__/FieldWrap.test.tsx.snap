// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`### FieldWrap FieldWrap renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: block;
  width: 100%;
}

<label
    class="emotion-0"
  />
</DocumentFragment>
`;
